keps/sig-api-machinery/0006-apply.md

```diff
@@ -301,58 +304,66 @@ The conversion between the two and creating the diff was complex and would have
-set the field manager for the parent.
+Currently, the admission chain for modifications with a subresource is not
+propagated through to the parent resource. This is important for tracking field
+ownership for resource-type subresources that modify the parent resource.
 
-For example, a validating webhook on Deployments to enforce a replica policy
-can be bypassed by any /scale subresource request because
-the webhook admission chain is called only for the subresource. We cannot track
-that the user who made the request to the /scale resource owns the replicas
-field.
+API requests to subresources are only intercepted by webhooks that registered
+for the subresource, not webhooks that registered for the parent resource.
+
+For example, a validating webhook on Deployments to enforce a replica policy can
+be bypassed by any /scale subresource request. The webhook admission chain is
+called only for the subresource, but not its parent. We cannot track that the
+user who made the request to the /scale resource owns the replicas field.
 
 ##### Proposed Change
 
 The webhooks will be called for the subresource and parent using the same
 admission context, such that the original user is represented in the admission
-attributes. Webhooks will opt-in to the new behavior (of being called for subresources, too) in the
-registration.
-
-The webhooks will be called in the following order, considering a request to
-`/foo/scale` as an example:
-1. mutating webhooks will be called on the included subresource (/foo/scale)
-2. mutating webhooks will be called on the parent (/foo)
-3. optionally, mutating webhooks will be called a second time in the same order
-   (1), then (2) (/foo/scale, then /foo)
-4. validating webhooks will be called in parallel for the subresource and parent
-   (/foo/scale and /foo simultaneously)
-
-Note that when applying mutating webhooks on the included subresources (1),
-then any modification attempts to implicitly modify fields of the parent that
-are not explicitly fields of the subresource will be discarded before writing to
+attributes. For compatibility, webhooks will opt-in to the new behavior (of
+being called for subresources, too) in the registration.
+
+The webhooks will be called in the order below, considering a request to
+`/foo/scale` as an example.
+
+1.  Mutating webhooks will be called on the included subresource (/foo/scale).
+1.  Mutating webhooks will be called on the parent. (/foo)
+1.  Optionally, mutating webhooks will be called a second time in the same order
+    (1), then (2) (/foo/scale, then /foo). We already do this for mutating
+    webhooks.
+1.  Validating webhooks will be called in parallel for the subresource and
+    parent (/foo/scale and /foo simultaneously). We already do this for
+    validating webhooks.
+
+Note that when applying mutating webhooks on the included subresources (1), then
+any modification attempts to implicitly modify fields of the parent that are not
```

Comment:
Was that what we agreed upon?

My thoughts are:
1. discarding without failing is usually a bad idea, since this is a "new feature" we don't have to do that for compatibility reasons. But that's a little muddy.
2. What's the rational for only updating the accessible fields? Right, one can't see the changes that happened since we don't return the full object. That makes me a little sad. That also sounds a little bit like a privilege escalation. Maybe you're right.

Reply:
No this wasn't agreed upon, but this is the strawman from this thread: https://github.com/kubernetes/kubernetes/issues/84530#issuecomment-577812497.